Set the desktopFileName property

fedelibre opened the pull request:
Fix #1248: wrong app id and missing icon in GNOME on
Wayland.

I found the solution [here](https://github.com/IJHack/QtPass/pull/468).
I started working on this just for fun but I don't know what I'm really doing :-)

Now the icon is appearing in the left dash on GNOME.
In the topbar the application name is correct, but there's no icon.. only a kind of stylised F.
Unless I miss something, this patch finally allows to find Frescobaldi in GNOME Software,  even though the icon is not found. See below screenshots:

![fresco-top-icon](https://user-images.githubusercontent.com/1644647/73600699-57278a80-4554-11ea-8ead-908cc0f99a4f.png)

![fresco-gnome-dash](https://user-images.githubusercontent.com/1644647/73600701-5d1d6b80-4554-11ea-91cf-3da15d66e050.png)

wbsoft commented:
Lgtm, although you can probably set the property on QApplication instead of QGuiApplication. Both are the same, QApplication inherits from QGuiApplication and is already imported.

fedelibre commented:
@wbsoft I noticed that these kind of properties were already set in app.py so I moved it there and made it more pythonic :-)

I've verified that it's now working fine both on X11 and on Wayland.
The only issue I see is the **missing icon in the top bar of GNOME shell**. If I find some time I'll try to investigate. If you don't see new commits in this PR in the next two days, you can merge it and we'll take care of the icon later (what we've already fixed is more important IMO).

fedelibre commented:
I've verified this branch in another computer and also with another user in the same laptop. It works fine in both cases, either running fresco from the repository or from the installation directory.
So my guess is that I have some stale files or other installations which are interfering (I did several local installations in my home directory while working on the flatpak for Frescobaldi).

For me this is Ok to be merged.

![fresco-topbar-ok](https://user-images.githubusercontent.com/1644647/73757811-143f0000-476a-11ea-9686-a4ddbf9a6d7c.png)

fedelibre commented:
>  In the topbar the application name is correct, but there's no icon.. only a kind of stylised F.

Ok, this is embarassing. I forgot I played with "symbolic icon discovering" some time ago.

![Schermata da 2020-02-05 00-08-03](https://user-images.githubusercontent.com/1644647/73795720-a1567900-47ab-11ea-8938-1bbe678c4e76.png)

fedelibre commented:
A drawback of this PR - or maybe #1249 - is that Frescobaldi will create and use a new configuration file `~/.config/org.frescobaldi.Frescobaldi.conf` without copying the contents of previous file `~/.config/frescobaldi.conf`. I had to do it manually.

fedelibre commented:
I think it would be desirable to keep `~/.config/frescobaldi/frescobaldi.conf`.
But if I set back the application name to `frescobaldi`, then in X11 the window manager will fail to find the desktop file.

This is what I get when application name is rdns_name:

```
$ xprop | grep WM_CLASS
WM_CLASS(STRING) = "frescobaldi", "org.frescobaldi.Frescobaldi"
```

and this is what I get when application name is name:

```
$ xprop | grep WM_CLASS
WM_CLASS(STRING) = "frescobaldi", "frescobaldi"
```

I wonder if there are other ways to change the WM_CLASS. Other than changing QApplication.setApplicationName.
Same problem is discussed for example [here](https://github.com/bitcoin/bitcoin/issues/10064#issuecomment-289019240).

I'm afraid this is going to be beyond my limited skills... let me know your thoughts.

wbsoft commented:
Settings really should be taken over. On windows the registry is used, no frescobaldi.conf is used there. Maybe there is a away to have QSettings just use the old application name etc. Will look into it.

We could change the settings location, install.py could then take care of that. But it's hairy, especially when users run multiple versions of Frescobaldi in their home directory.

wbsoft commented:
It is possible to construct QSettings with its own application and organisation name.
https://doc.qt.io/qt-5/qsettings.html
I will devise a way to do that throughout Frescobaldi.

PeterBjuhr commented:
@fedelibre what is the status of this?

fedelibre commented:
I'm stuck, because this patch is also changing the name of the configuration file (at least in Linux) and we don't want this. Ideally, I would like to use the reverse DNS name for the application name and the normal name for the configuration file. But I couldn't find an easy way to do it.

fedelibre commented:
After checking what others did, I think that the desktop file name should include the .desktop extension so I should probably change a line to:

    QApplication.setDesktopFileName('org.frescobaldi.Frescobaldi.desktop')

I don't have much time and energy to try to understand this issue.

fedelibre commented:
<Face palm> It seems it's easier than expected. Setting the desktop file seems to be enough, no need to change the application name, since Wayland finds the right app ID in the desktop file.

I'm going to open a new PR and eventually we'll close this one if the fix is confirmed.

fedelibre commented:
replaced by #1478
